### PR TITLE
accidentally merged already -- adding change to make pull request

### PR DESCRIPTION
**Pull request by ehwgal**
# Pull request guidelines

Welcome to the 🐸STT project! We are excited to see your interest, and appreciate your support!

This repository is governed by the Contributor Covenant Code of Conduct. For more details, see the [CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md) file.

In order to make a good pull request, please see our [CONTRIBUTING.rst](CONTRIBUTING.rst) file, in particular make sure you have set-up and run the pre-commit hook to check your changes for code style violations.

Before accepting your pull request, you will be asked to sign a [Contributor License Agreement](https://cla-assistant.io/coqui-ai/STT).

This [Contributor License Agreement](https://cla-assistant.io/coqui-ai/STT):

- Protects you, Coqui, and the users of the code.
- Does not change your rights to use your contributions for any purpose.
- Does not change the license of the 🐸STT project. It just makes the terms of your contribution clearer and lets us know you are OK to contribute.

### CODE REVIEW

**Comment by ehwgal** [comment]
@s1nayak

@H-Matuszewski @vickyivova 

Here is my project, looking forward to your review!

Just a sidenote: I accidentally merged my branch with the main branch already, so I just implemented an unimportant change to be able to make this pull request.

**Comment by H-Matuszewski** [comment]
@s1nayak

* The notebook runs smoothly. While it takes a long time to train and whatnot, it's merely a case of leaving it to run and then coming back to it.
* Having the pip installs in the notebook is nice as there isn't any issue of running a requirements document.
* The cells were split up nicely so that there wasn't too much going on per cell, whilst also not being littered with too few.
* Never heard of gdown before but it seems like a very smart way of avoid the need to download, upload and just generally move files around. Just the notebook, and a clean execution.<br>
<br>
**Potential points of improvement:**<br>
_Note - I wouldn't say it's necessary to incorporate all of these, I'm not even sure to what extent some of the aspects can be controlled by the user and what is intrinsic to COQUI._<br>
<br>
<br>
* Perhaps it might be a nice idea to have the Assignment README in the folder of the assignment. The original README of the project also probably contains a lot of useful information. Rather than removing it, you can avoid repeating yourself with certain things and focus more on the task at hand.<br>
* The amount of printing for the Best, Median and Worst WER seems excessive, I think it could do with a lot less. Perhaps just 1 for each category?<br>
* Some visualisations would be nice here. Might it be possible to retrieve all values as a list/numpy and do something along the lines of a seaborn distplot? If comparing for over multiple datasets, perhaps a histogram of each result (average over a dataset)?<br>
* For sharing purposes it may be a good idea to set the amount of epochs to 1, so that the user can have a quicker "first run" of the code. Once they get comfortable (and also know how long an epoch takes, roughly), they can adjust accordingly. In my opinion (however unqualified it may be) it seems better to optimise speed of execution rather than robustness of results for shared versions of code.<br>
* From broader data science analysis, it'd be cool to label the data in some sort of way and do further analysis. Perhaps dividing the dataset into a boolean of HasName, or dividing them up by the amount of words contained in each sample, and so on. Comparing these against each other (so let's say how much the effectiveness of the trained model given that a name is present in the sample) might yield interesting insights and may give indications on how to further train it.<br>
<br>
<br>
Overall I like the intuitiveness and structure of the program, and it's ease of use and convenience are a major bonus as well. I could easily see myself training my own model whilst using this as a starting point.

**Comment by ehwgal** [comment]
> @s1nayak
> 
>     * The notebook runs smoothly. While it takes a long time to train and whatnot, it's merely a case of leaving it to run and then coming back to it.
> 
>     * Having the pip installs in the notebook is nice as there isn't any issue of running a requirements document.
> 
>     * The cells were split up nicely so that there wasn't too much going on per cell, whilst also not being littered with too few.
> 
>     * Never heard of gdown before but it seems like a very smart way of avoid the need to download, upload and just generally move files around. Just the notebook, and a clean execution.
> 
> 
> 
> **Potential points of improvement:**
> _Note - I wouldn't say it's necessary to incorporate all of these, I'm not even sure to what extent some of the aspects can be controlled by the user and what is intrinsic to COQUI._
> 
> 
> * Perhaps it might be a nice idea to have the Assignment README in the folder of the assignment. The original README of the project also probably contains a lot of useful information. Rather than removing it, you can avoid repeating yourself with certain things and focus more on the task at hand.
> * The amount of printing for the Best, Median and Worst WER seems excessive, I think it could do with a lot less. Perhaps just 1 for each category?
> * Some visualisations would be nice here. Might it be possible to retrieve all values as a list/numpy and do something along the lines of a seaborn distplot? If comparing for over multiple datasets, perhaps a histogram of each result (average over a dataset)?
> * For sharing purposes it may be a good idea to set the amount of epochs to 1, so that the user can have a quicker "first run" of the code. Once they get comfortable (and also know how long an epoch takes, roughly), they can adjust accordingly. In my opinion (however unqualified it may be) it seems better to optimise speed of execution rather than robustness of results for shared versions of code.
> * From broader data science analysis, it'd be cool to label the data in some sort of way and do further analysis. Perhaps dividing the dataset into a boolean of HasName, or dividing them up by the amount of words contained in each sample, and so on. Comparing these against each other (so let's say how much the effectiveness of the trained model given that a name is present in the sample) might yield interesting insights and may give indications on how to further train it.
> 
> 
> Overall I like the intuitiveness and structure of the program, and it's ease of use and convenience are a major bonus as well. I could easily see myself training my own model whilst using this as a starting point.

-----------------------------------------------------
@s1nayak

**MY GRADE TO HUBERT: 10**

Hubert's feedback is clear and completely fair; I especially agree with him that I could shorten the print output for the WER. I actually wanted to write this output to a .txt file to make it a bit less cluttered, but I had not gotten to it yet. I also appreciate he listed things he liked about my project next to suggestions for improvement.

I agree that decreasing the number of epochs would be better, though I think 1 epoch would not be enough to have adequate results. Maybe I should specify the time it takes to run the model beforehand and indicate how to decrease the number of epochs to the user.

I am also not sure to what extent I could implement some of his suggestions, but I think they are good points. In relation to his last suggestion, I think maybe a challenge set (often used in NLP, where you manually compile a dataset comprised of things you suspect your model could have difficulty with, designed to more extensively probe the model) could work for an STT model like this as well.

**Comment by vickyivova** [comment]
Hey Ellemijn (@ehwgal), Hubert (@H-Matuszewski) and Shekhar (@s1nayak)!
--------------------------------------------------------------------------------------

I had some trouble finishing the training of your model due to issues on my part. But I managed to eventually! 
You do have quite some more data then me (double the amount of recordings) so as expected, the finetuning did take longer. However, you also have better results! Your loss is substantially less -- for me it ended up being around 27.0 and for you it is only 10.611! I wonder whether this is an effect only of the amount of data or also the quality (mine is quite echo-y and my accent is quite prominent). It would have been fun to compare our results if we had the same amount of data!

Your notebook ran very smoothly and the process of replicating your results was very streamlined. I will definitely adopt using the ```gdown``` package as it made downloading the data very easy. I needed to include mounting of Google Drive in my assignment and it is a bit of a hassle for the user.

As opposed to Hubert, I enjoy seeing the examples for Best, Median and Worst WER 😆. I do recognize that the printing looks a bit disorienting and like a lot of text but it is very satisfying to see exactly how your model does.

I do agree that keeping the original README could have been useful -- I did the same and I now see that it would have been better if I had a separate README for the assignment.

From the discussions we had while we were preparing the assignment I know that you needed to specify the alphabet as an argument to the initializations of the training as you otherwise got nonsensical results. I did not need to do that and still had no issue with the results. I wonder what that is about 🤔. I notice that the ```alphabet.txt``` file in your ```my-model``` folder has the apostrophe at the end of the alphabet while the ```alphabet.txt``` in the preloaded model folder starts with the apostrophe. For me that is not the case! Both files (from pretrained model and from my model) end with the apostrophe. Interesting! I find it a little bit hard to figure out exactly how the ```alphabet.txt``` is generated but it is interesting to look at!

Overall, super fun to see your approach to the assignment! ❤️

**Comment by ehwgal** [comment]
> ## Hey Ellemijn (@ehwgal), Hubert (@H-Matuszewski) and Shekhar (@s1nayak)!
> 
> I had some trouble finishing the training of your model due to issues on my part. But I managed to eventually! You do have quite some more data then me (double the amount of recordings) so as expected, the finetuning did take longer. However, you also have better results! Your loss is substantially less -- for me it ended up being around 27.0 and for you it is only 10.611! I wonder whether this is an effect only of the amount of data or also the quality (mine is quite echo-y and my accent is quite prominent). It would have been fun to compare our results if we had the same amount of data!
> 
> Your notebook ran very smoothly and the process of replicating your results was very streamlined. I will definitely adopt using the `gdown` package as it made downloading the data very easy. I needed to include mounting of Google Drive in my assignment and it is a bit of a hassle for the user.
> 
> As opposed to Hubert, I enjoy seeing the examples for Best, Median and Worst WER 😆. I do recognize that the printing looks a bit disorienting and like a lot of text but it is very satisfying to see exactly how your model does.
> 
> I do agree that keeping the original README could have been useful -- I did the same and I now see that it would have been better if I had a separate README for the assignment.
> 
> From the discussions we had while we were preparing the assignment I know that you needed to specify the alphabet as an argument to the initializations of the training as you otherwise got nonsensical results. I did not need to do that and still had no issue with the results. I wonder what that is about 🤔. I notice that the `alphabet.txt` file in your `my-model` folder has the apostrophe at the end of the alphabet while the `alphabet.txt` in the preloaded model folder starts with the apostrophe. For me that is not the case! Both files (from pretrained model and from my model) end with the apostrophe. Interesting! I find it a little bit hard to figure out exactly how the `alphabet.txt` is generated but it is interesting to look at!
> 
> Overall, super fun to see your approach to the assignment! ❤️

-------------------------------------------
@s1nayak

**MY GRADE TO VICKY: 10**

Thank you for your kind words Vicky!

The alphabet.txt file is indeed an issue I need to research further, as I needed to specify that I was using the alphabet.txt file of the pre-trained model rather than that of my own model. One of my suspicions is that there is somehow a character in my test set that does not appear in my training set, or some of the encoding could be different, although I am not quite sure whether this is how it works / whether this is how alphabet.txt is generated.

Keeping the original README would have probably been useful. You can still view the README in the project under README**.srt** (instead of README**.md**) as specified, but this approach might be a little illogical so I understand where you and Hubert are coming from. 

I am not sure the ``gdown`` package is any better than regularly downloading data files and then mounting, as you do need to make sure the URLs are working and have space to store the files on your Drive, but it was a fun experiment! It's especially good for large files that which do not work with 'wget' or 'curl' -- at first I had an incredibly large dataset, which is why I was forced to use ``gdown``, but my current dataset is small enough for wget or curl to have worked.
